### ISSUE TITLE
Getting errors when trying to run job with a worker model, from my own image

### ISSUE THREAD

**Issue by odedsCoolScripts** [ovh/cds#4285]
Hi Everyone,

I'll describe my configuration and explain my issue below:

**Configuration:**

1. I'm running Kuberentes on AWS EKS. The cluster functions as expected: running jobs, containers from private AWS registry, etc..
2. Installed CDS on K8s successfully, using the CDS repo, cloned to my Mac and Helm.
3. In CDS UI, crated a workspace, pipline and inside a pipline created a job for CDS education.
4. Created a worker model with image from our private ECR. I think that CDS downloaded the image successfully because I see that it passed that 'curl' requirement test, when creating the module. Any way, without CDS, our K8s cluster (AWS EKS), authenticate the ECR (registry) by access role, configured in AWS IAM and It downloads images from our private ECR smoothly. 
5. In my Mac, installed 'cdsctl', and connected successfully to my CDS. Verified with cdsctl health status
6. In another workflow I could install a public Go docker image (for sanity test) on my CDS system and run a job successfully.

**The issue:**
 in continuance to section 4 above, I tried to run a very simple job, associated with the requirement with a worker model, from my own image and the job never ends.
I copied the logs I found in the Hatchery pod container:

```
 2019-05-13 14:51:23 [DEBUG] hatchery> workerRegister> WorkerModelBook on model tekapp-new err: Serve>CommonServe>Create>workerRegister>WorkerModelBook: Worker Model already booked (caused by: Error)
2019-05-13 14:52:23 [DEBUG] hatchery> workerRegister> need register
2019-05-13 14:52:23 [DEBUG] hatchery> checkCapacities> 0.000 seconds elapsed
2019-05-13 14:52:23 [DEBUG] hatchery> workerRegister> WorkerModelBook on model tekapp-new err: Serve>CommonServe>Create>workerRegister>WorkerModelBook: Worker Model already booked (caused by: Error)
	2019-05-13 14:53:23 [DEBUG] hatchery> workerRegister> need register
2019-05-13 14:53:23 [DEBUG] hatchery> checkCapacities> 0.000 seconds elapsed
2019-05-13 14:53:23 [DEBUG] hatchery> workerRegister> WorkerModelBook on model tekapp-new err: Serve>CommonServe>Create>workerRegister>WorkerModelBook: Worker Model already booked (caused by: Error)
2019-05-13 14:54:17 [DEBUG] hatchery> checkCapacities> 0.000 seconds elapsed
2019-05-13 14:54:17 [DEBUG] canRunJob> model tekapp-new needs registration
2019-05-13 14:54:17 [DEBUG] canRunJob> 1557759257 - job 35 - model requirement r.Value(tekapp-new) != model.Name(tutum-curl)
2019-05-13 14:54:17 [DEBUG] hatchery> no model
2019-05-13 14:54:23 [DEBUG] hatchery> workerRegister> need register
2019-05-13 14:54:23 [DEBUG] hatchery> checkCapacities> 0.000 seconds elapsed
2019-05-13 14:54:23 [DEBUG] hatchery> workerRegister> WorkerModelBook on model tekapp-new err: Serve>CommonServe>Create>workerRegister>WorkerModelBook: Worker Model already booked (caused by: Error)
2019-05-13 14:54:23 [DEBUG] job 35 already spawned in previous routine
2019-05-13 14:54:43 [DEBUG] hatchery> checkCapacities> 0.000 seconds elapsed
2019-05-13 14:54:43 [DEBUG] canRunJob> model tekapp-new needs registration
2019-05-13 14:54:43 [DEBUG] canRunJob> 1557759283 - job 35 - model requirement r.Value(tekapp-new) != model.Name(tutum-curl)
2019-05-13 14:54:43 [DEBUG] hatchery> no model
2019-05-13 14:55:03 [DEBUG] hatchery> checkCapacities> 0.000 seconds elapsed
2019-05-13 14:55:03 [DEBUG] canRunJob> model tekapp-new needs registration
2019-05-13 14:55:03 [DEBUG] canRunJob> 1557759303 - job 35 - model requirement r.Value(tekapp-new) != model.Name(tutum-curl)
2019-05-13 14:55:03 [DEBUG] hatchery> no model
```

**Comment by yesnault** [comment]
I guess: 
- Your configure the model `tekapp-new ` with an image from private registry.
- The model `tutum-curl` is the test from a public image

But, private registry is supported on version 0.40.x (next release). Which cds version do you use? (if master of today, should be ok, otherwise, it probably won't work)

**Comment by odedsCoolScripts** [comment]
Thanks a lot @yesnault 
You're guess is correct


Is this command:
`cdsctl version` telling what's the CDS system version I'm running? 
I can see in the Helm values.yaml that my version is 0.39.2

Is there any documentation of how to update the CDS system using Helm?

Thanks,

Oded

**Comment by yesnault** [comment]
about version, it's the latest, it's ok.
about upgrading helm: `helm upgrade`. But I think there is missing wrong about helm upgrade command with the password regenerated on the fly. https://github.com/ovh/cds/issues/4286 to follow this point.
About current issue: it's linked to the new feature that it will be released on 0.40.0. Feel free to re-open an issue if there any error with this release. 0.40.0 will be the next release.
